### DISCUSSION TITLE
Merging Datasets

### DISCUSSION THREAD

**Discussion by beroshcr** [mtex-toolbox/mtex#1636] [Ask Anything]
Hello,

I have two separate EBSD datasets analyzed in the same identical area of my sample. The difference between datasets is simply the minerals indexed. Is there a way to combine the datasets? I have the data in both .cpr and .crc files.

![Christain-June-3-2022-Specimen-1-Area-1-Run-2-Map-Data-2-Ph](https://user-images.githubusercontent.com/57849453/200933520-2bd22a53-2f31-4d83-8579-e5c275f29990.jpeg)

![Christain-June-3-2022-Specimen-1-Area-1-Run-2-Reanalyzed-Map-Data-11-Ph](https://user-images.githubusercontent.com/57849453/200933535-974a8a36-026f-4fac-832b-56aa4cf3f8be.jpg)

Thanks!

**Comment by kilir** [comment]
Hi,
assuming they are ion the same grid, you can do something like the following:
```
mtexdata sharp

prop.x = ebsd.prop.x
prop.y = ebsd.prop.y
phaseId = ebsd.phaseId;
rots = ebsd.rotations;
CSList=ebsd.CSList;


% replace some points e.g.
% which could come from a different map

CSList{end+1}=crystalSymmetry('1','Mineral','NewPhase')

idofnewPhase = randi(length(phaseId),500,1)   % <-put something useful here, e.g. indices from your 2nd map
rotnew = rotation.rand(500)                                % <-put something useful here, e.g. the rotations from the indices from your 2nd map

phaseId(idofnewPhase)=3;
rots(idofnewPhase)= rotnew;

ebsdnew = EBSD(rots,phaseId,CSList,prop)
```
Cheers,
Rüdiger

**Comment by beroshcr** [comment]
Hi Rüdiger, 

Thank you for your reply and help so far. Both data sets are on the same grid. To explain this further, I'd like to overlay and merge both datasets together to attempt to fill in the gaps of non-indexed pixels. 

Could you perhaps walk me through this bit of code?
```
idofnewPhase = randi(length(phaseId),500,1)   % <-put something useful here, e.g. indices from your 2nd map
rotnew = rotation.rand(500)                                % <-put something useful here, e.g. the rotations from the indices from your 2nd map
```
What does "rand" signify? What does the quantity of "500" signify? I am a little confused. Because I have two separate datasets how do I go about inputting this information? 

Christian

**Reply by kilir** [reply]
Hi,
these two lines are about to be irrelevant for your case and are just there to demonstrate how to change something on these lists.

You should find a) the indices of the list of phases which you want to replace and b) create a list containing the rotations of those.

Roughly for example
```
ind = ebsd1.phaseId==0 & ebsd2.phaseId == 3
```
which would mean in this case the pixels you want to find are in ebsd1 those which have phaseId 0 (in most datasets that's notIndexed) AND in the second dataset ebsd2 it should be the fourth phase (counting from 0). You need to adapt that for your case.

next you'd want the list of orientations
```
rotnew = ebsd2(ind).rotations;
```
And set a new phase Id according to the updated CSList and rotations as in the previous example.
```
phaseId(ind)=3;
rots(idofnewPhase)= rotnew;
```
.
Hope that helps.

Cheers,
Rüdiger

**Reply by beroshcr** [reply]
Hi Rüdiger,

Thank you once again. I am still learning how to code so I appreciate the details. I'm going to try and figure this out and if any other questions arise I'll reply again in the thread. 

Christian
